frontend/src/modules/_shared/LayerFramework/settings/implementations/BooleanSetting.tsx

```diff
@@ -0,0 +1,31 @@
+import type { ChangeEvent } from "react";
+import type React from "react";
+
+import { Switch } from "@lib/components/Switch";
+
+import type {
+    CustomSettingImplementation,
+    SettingComponentProps,
+} from "../../interfacesAndTypes/customSettingImplementation";
+import type { SettingCategory } from "../settingsDefinitions";
+
+type ValueType = boolean;
+export class BooleanSetting implements CustomSettingImplementation<ValueType, SettingCategory.BOOLEAN> {
+    isValueValid(): boolean {
+        return true;
+    }
+
+    getIsStatic(): boolean {
+        return true;
+    }
+
+    makeComponent(): (props: SettingComponentProps<ValueType, SettingCategory.BOOLEAN>) => React.ReactNode {
+        return function BooleanSwitch(props: SettingComponentProps<ValueType, SettingCategory.BOOLEAN>) {
+            function handleChange(e: ChangeEvent<HTMLInputElement>) {
+                props.onValueChange(e.target.checked);
+            }
+
+            return <Switch checked={props.value} onChange={handleChange} />;
+        };
+    }
```

Comment:
Just a thought: Could consider providing an enum in constructor of the `BooleanSetting` which allows us so select between `Switch` and `Checkbox` for a boolean setting, and then render/make component using the corresponding component from lib?